lib/charms/mongodb/v0/config_server_interface.py

```diff
@@ -286,6 +296,34 @@ def _on_relation_broken(self, event: RelationBrokenEvent) -> None:
         self.charm.remove_connection_info()
 
     # BEGIN: helper functions
+    def pass_hook_checks(self, event):
+        """Runs the pre-hooks checks for ClusterRequirer, returns True if all pass."""
+        if self.is_mongos_tls_needed():
+            logger.info(
+                "Deferring %s. Config-server uses TLS, but mongos does not. Please synchronise encryption methods.",
+                str(type(event)),
+            )
+            event.defer()
```

Comment:
nit: given this and all subsequent conditions evaluating to false trigger an event deferral, would it make sense to move the event.defer to the hook itself ?

Reply:
we could, but in the future we may need to change it back, if an event warrants not needing to defer.

We have other `pass_hook` functions in other areas of the code for example `lib/charms/mongodb/v1/sharding.py` and they are responsible for defering events. I would rather have code consistency.

If you feel strongly though I am more than happy to, just let me know